packages/staking/src/features/store/useDelegationPortfolioStore.ts

```diff
@@ -0,0 +1,231 @@
+import { DelegatedStake } from '@cardano-sdk/wallet';
+import { Wallet } from '@lace/cardano';
+import { create } from 'zustand';
+import { immer } from 'zustand/middleware/immer';
+import { MAX_POOLS_COUNT } from './delegationPortfolio';
+import { CurrentPortfolioStakePool, DraftPortfolioStakePool } from './types';
+
+enum Mode {
+  Overview = 'Overview',
+  BrowsePools = 'BrowsePools',
+  CurrentPoolDetails = 'CurrentPoolDetails',
+  PoolDetails = 'PoolDetails',
+  CurrentPortfolioManagement = 'CurrentPortfolioManagement',
+  NewPortfolioCreation = 'NewPortfolioCreation',
+}
+
+enum DrawerDefaultSection {
+  DETAIL = 'detail',
+}
+
+enum DrawerManagementSection {
+  PREFERENCES = 'preferences',
+  CONFIRMATION = 'confirmation',
+  SIGN = 'sign',
+  SUCCESS_TX = 'success_tx',
+  FAIL_TX = 'fail_tx',
+}
+
+type Section = DrawerDefaultSection | DrawerManagementSection;
+
+enum OverviewCommand {
+  ShowDetails = 'ShowDetails',
+  Manage = 'Manage',
+  GoToBrowsePools = 'GoToBrowsePools',
+}
+
+enum BrowsePoolsCommand {
+  Select = 'Select',
+  ShowDetails = 'ShowDetails',
+}
+
+enum PoolDetailsCommand {
+  Select = 'Select',
+}
+
+enum DrawerPreferencesCommand {
+  UpdateWeight = 'UpdateWeight',
+}
+
+type Command = OverviewCommand | BrowsePoolsCommand | PoolDetailsCommand | DrawerPreferencesCommand;
```

Comment:
I'd consider making this an interface and be explicit in the commands that these are implementing this interface, which would then have the benefit of ensuring consistency of the structure and should in theoery allow even for extracting the individual Commands into separate files and generalize the handler in a way that adding a new Command wouldn't force the generic logic to be updated

```suggestion
interface Command = { name: string, params: unknown };
```

Reply:
note, actually an ordinary `type` should do, an interface may mislead the reader into thinking that classes are involved which isn't the case